ffcli/command_test.go

```diff
@@ -439,6 +439,65 @@ func TestIssue57(t *testing.T) {
 	}
 }
 
+func TestSharedFlags(t *testing.T) {
+	for _, testcase := range []struct {
+		name       string
+		args       []string
+		sharedFlag string
+	}{
+		{
+			name:       "foo",
+			args:       []string{"foo"},
+			sharedFlag: "foo-default",
+		},
+		{
+			name:       "bar",
+			args:       []string{"bar"},
+			sharedFlag: "bar-default",
+		},
+		{
+			name:       "foo -baz=test",
+			args:       []string{"foo", "-baz=test"},
+			sharedFlag: "test",
+		},
+		{
+			name:       "bar -baz=test",
+			args:       []string{"bar", "-baz=test"},
+			sharedFlag: "test",
+		},
+		{
+			name:       "foo -baz=",
+			args:       []string{"foo", "-baz="},
+			sharedFlag: "",
+		},
+		{
+			name:       "bar -baz=",
+			args:       []string{"bar", "-baz="},
+			sharedFlag: "",
+		},
+	} {
+		t.Run(testcase.name, func(t *testing.T) {
+			sharedFlag := "init-value"
+			fooFs := flag.NewFlagSet("foo", flag.ContinueOnError)
+			barFs := flag.NewFlagSet("bar", flag.ContinueOnError)
+			fooFs.StringVar(&sharedFlag, "baz", "foo-default", "testing")
+			barFs.StringVar(&sharedFlag, "baz", "bar-default", "testing")
```

Comment:
Haha! This is a bug (?) with stdlib flag.FlagSet. 

The default value provided to e.g. fs.StringVar is assigned to the associated variable _at declaration time_, i.e. when you call fs.StringVar, and _not_ when you call fs.Parse. So the default value you see will be whichever value is given in the final fs.StringVar call you make — in your test, it's this one.

I think the only fix here would be, somehow, avoiding assigning a default value in the declaration, and populating it later. Or defining a separate e.g. DeferredFlagVar type that encapsulated this behavior? I don't know offhand what's best. Ugh.

Reply:
I can confirm it's not related with ff :)

```go
sharedFlag := "init-value"
fooFs := flag.NewFlagSet("foo", flag.ContinueOnError)
barFs := flag.NewFlagSet("bar", flag.ContinueOnError)
fooFs.StringVar(&sharedFlag, "baz", "foo-default", "testing")
barFs.StringVar(&sharedFlag, "baz", "bar-default", "testing")
_ = fooFs.Parse([]string{""})
fmt.Println(*sharedFlag) // bar-default
```

I'll ask on golang-nuts if I need to report a bug and see if I can make a contribution on the go codebase :)